acme/acme/messages.py

```diff
@@ -315,7 +315,7 @@ class Registration(ResourceBody):
     # on new-reg key server ignores 'key' and populates it based on
     # JWS.signature.combined.jwk
     key = jose.Field('key', omitempty=True, decoder=jose.JWK.from_json)
-    contact = jose.Field('contact', omitempty=True, default=())
+    contact = fields.UnFalseyField('contact', omitempty=True)
```

Comment:
We're close here but there are some things I don't like about this approach. In particular, it changes the default value of the field and there are currently problems like:
```
In [3]: Registration().emails
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-48d171e81d05> in <module>
----> 1 Registration().emails

acme/acme/messages.py in emails(self)
    354     def emails(self):
    355         """All emails found in the ``contact`` field."""
--> 356         return self._filter_contact(self.email_prefix)
    357 
    358 

acme/acme/messages.py in _filter_contact(self, prefix)
    342 
    343     def _filter_contact(self, prefix):
--> 344         return tuple(
    345             detail[len(prefix):] for detail in self.contact  # pylint: disable=not-an-iterable
    346             if detail.startswith(prefix))

TypeError: 'NoneType' object is not iterable
```
We could fix the latter, but I would like to keep the default the same which will automatically fix this problem. Doing that is much harder though.

I didn't implement this so there may be problems that I'm not seeing right now, but to resolve https://github.com/certbot/certbot/issues/5129, I think we should take an approach similar to what we did in `acme/mixins.py`. I would make the following changes to `messages.UpdateRegistration`:

1. Add an `__init__` method that checks if `contact` was included in the given arguments, setting a private (prefixed with an underscore) boolean variable with the result. See `mixins.py` about how to set this attribute.
2. Add `to_partial_json` and `fields_to_partial_json` methods that call the parent method and removes the `contact` field if it wasn't set by the user and just has it's default value.

This is certainly more complex and less elegant than what was done here, however, I think it reduces the amount of unintended changes to behavior made in this PR which I think is pretty valuable.